clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/AccountHealthIndicator.java

```diff
@@ -32,11 +35,30 @@
  * bringing down all of clouddriver, but any errors associated with individual accounts will appear
  * in the detailed health information.
  *
+ * <p>The number of unhealthy accounts will be reported as the metric health.id.errors, where id is
+ * the id supplied to the constructor.
+ *
  * @param <T> The type of account credentials this health indicator supports
  */
 public abstract class AccountHealthIndicator<T extends AccountCredentials>
     implements HealthIndicator {
   @Nonnull private volatile Health health = new Health.Builder().up().build();
+  @Nonnull private final AtomicLong unhealthyAccounts = new AtomicLong(0);
+
+  /**
+   * Create an {@code AccountHealthIndicator} reporting metrics to the supplied registry, using the
```

Comment:
This might be a stupid question, my brian is :hankey: today. What's an example of a "supplied registry?" Where are these health metrics going?

Reply:
This is a `com.netflix.spectator.api.Registry`, which allows you to register data points that you want to track and report metrics on.  So in this case by calling `PolledMeter.using(registry).withName("name").monitorValue(value)` we're telling the registry to keep track of the number behind the reference that `value` points to.  (So it has to be some reference rather than a primitive since it needs to hold a constant reference to monitor and will periodically look and record the value behind that reference.)

Then if you send a GET request to `http://localhost:7002/spectator/metrics` you'll get a data dump of all the metrics that clouddriver knows about.  I think that most integrations around monitoring effectively poll that endpoint and feed the data into monitoring systems.  (The monitoring daemon for example will poll that endpoint and handle pushing to any systems you have.)

So I guess in summary, here `Registry` is the global registry of all metrics that clouddriver has (which is a bean that is created in the base config) and we need it to be able to register another metric.

Reply:
Thank you!